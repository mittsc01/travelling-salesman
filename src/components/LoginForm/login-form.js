import React from 'react'
import RaceContext from '../../LoginContext'
import AuthApiService from '../../services/auth-api-service'


export default class LoginForm extends React.Component {
    

    state = { error: null }
    static contextType = RaceContext
    handleSubmitJwtAuth = ev => {
      ev.preventDefault()
      this.setState({ error: null })
      const { user_name, password } = ev.target
  
      AuthApiService.postLogin({
        user_name: user_name.value,
        password: password.value,
      })
        .then(res => {
          user_name.value = ''
          password.value = ''
          this.context.handleLogin()
          this.props.history.push('/schedule')
        })
        .catch(res => {
          alert(res.error)
          
          this.setState({ error: res.error })
        })
    }

    render(){
        return (<form className='login-form' onSubmit={this.handleSubmitJwtAuth}>
          <div className="form-div">
          <label htmlFor="user_name">username</label>
            <input type='text' placeholder="username" name="user_name"></input>
            <label htmlFor="password">password</label>
            <input type="password" placeholder="password" name="password"></input>
            
          </div>
          <button className="save-button" type="submit">Log in</button>
            
        </form>)
    }
}