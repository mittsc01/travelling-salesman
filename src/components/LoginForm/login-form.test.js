import LoginForm from './login-form';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom'
it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<LoginForm />, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<LoginForm />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });