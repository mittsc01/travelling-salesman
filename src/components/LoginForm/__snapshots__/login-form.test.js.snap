// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<form
  className="login-form"
  onSubmit={[Function]}
>
  <div
    className="form-div"
  >
    <label
      htmlFor="user_name"
    >
      username
    </label>
    <input
      name="user_name"
      placeholder="username"
      type="text"
    />
    <label
      htmlFor="password"
    >
      password
    </label>
    <input
      name="password"
      placeholder="password"
      type="password"
    />
  </div>
  <button
    className="save-button"
    type="submit"
  >
    Log in
  </button>
</form>
`;
